stl/inc/numeric

```diff
@@ -461,6 +461,24 @@ _FwdIt2 transform_inclusive_scan(_ExPo&& _Exec, _FwdIt1 _First, _FwdIt1 _Last, _
     _UnaryOp _Transform_op) noexcept; // terminates
 #endif // _HAS_CXX17
 
+template <class _Ty>
+bool _Arrays_overlap(const _Ty* const _First, const _Ty* const _Second, const size_t _Common_count) {
+    const uintptr_t _First_val  = reinterpret_cast<uintptr_t>(_First);
+    const uintptr_t _Second_val = reinterpret_cast<uintptr_t>(_Second);
+    const size_t Size_bytes     = _Common_count * sizeof(_Ty);
+    return _First_val + Size_bytes > _Second_val && _Second_val + Size_bytes > _First_val;
+}
+
+template <class _Ty, class _BinOp>
+void _Adjacent_difference_no_overlap(
+    _Ty* __restrict _Dest, const _Ty* __restrict _Src, const size_t _Count, _BinOp _Func) {

```

Comment:
Cv-qualification possibly matters, should we keep the original cv-qualification?

Reply:
Good point. Also I'll add top-level `const` to pointers.

Reply:
Ah, I forgot saying that there should be 2 different template parameters if do so...